app/src/main/java/com/github/sdp_begreen/begreen/models/event/Event.kt

```diff
@@ -19,4 +19,6 @@ interface Event<T>: CopyableWithId<T> {
      * The root path is used to know where to store the event on the database
      */
     val rootPath: RootPath
+
+    override fun equals(other: Any?): Boolean
```

Comment:
I don't really understand what you mean by `it doesn't force the classes that implement it to override its body`?

But then for the reason of having this method present here, it is because int the file [`EventsListAdapter`](https://github.com/SDP-BeGreen/BeGreen/blob/e984da5b556774e6669d1ba65982780f112ba3fd/app/src/main/java/com/github/sdp_begreen/begreen/adapters/EventsListAdapter.kt#L78), in the method `areContentsTheSame`, I have to compare two `Event` by equality. So I have to explicitly tell that the interface `Event` has the `equals` method. Otherwise it is not possible to actually compare them.

And then as that's only `data class` that implement this method, the `equals` method is already overridden by default.

And then I didn't explicitly implemented the `hashcode` method in the interface, because I never actually use it, and as `Event`'s subclasses are all `data class`, the `hashcode` metod is also already implemented, if one of those class actually need it.

But then again I can add it directly to the interface of course. Do you  think it would be better to add it ?

Reply:
Thank you for your explanations, it's clear. Since `equals()` is not overriden, we can omit the `hashcode()` method. If someone wants to override the `equals()` method, he will in any case be responsible for overriding the `hashcode()` method. So it's fine to keep it like that.